### ISSUE TITLE
Reading a cookie value results in the cookie beeing set again

### ISSUE THREAD

**Issue by david-plugge** [elysiajs/elysia#857]
### What version of Elysia is running?

1.1.17

### What platform is your computer?

Linux 5.15.153.1-microsoft-standard-WSL2 x86_64 unknown

### What steps can reproduce the bug?

When reading a cookie that is present a `set-cookie` header is added to the response, even when no new cookie value is set.

To reproduce just create a new Elysia project (`bun create elysie`) and copy the following code into `src/index.ts`
```ts
import { Elysia } from 'elysia';

new Elysia()
	.get('/', () => 'Hello Elysia')

	.get('/set', ({ cookie: { test } }) => {
		test.value = Math.random().toString();

		return {
			value: test.value,
		};
	})
	.get('/get', ({ cookie: { test } }) => {
		return {
			value: test.value,
		};
	})

	.listen(3000, (server) => {
		console.log(`🦊 Elysia is running at ${server.url}`);
	});
```

Steps to reproduce:
1. visit `http://localhost:3000/get` -> `set-cookie` header is not present
2. visit `http://localhost:3000/set` -> `set-cookie` header is present
3. visit `http://localhost:3000/get` -> `set-cookie` header is present

### What is the expected behavior?

The cookie is not set again

### What do you see instead?

The cookie is set again.

### Additional information

The session cookie i use in my session service gets set over and over. That itself is not that big of a deal in terms of functionality, but due to missing informations about the cookie lifetime for example the cookie is always set using the default options (no path, no domain, no secure, no httpOnly, ...).

### Have you try removing the `node_modules` and `bun.lockb` and try again yet?

yes

**Comment by duydang2311** [comment]
I'm also having this issue causing my app to log out every time the cookie is modified invalidly. The Set-Cookie header should only be set if either `my_cookie.set(...)` or `my_cookie.value` setter was called.

**Comment by Kneemund** [comment]
I can confirm this. Using the JWT plugin will reset the cookie attributes when doing any request after initially setting the token.

**Comment by qisoft** [comment]
I'm having the same

**Comment by qisoft** [comment]
I believe this was caused by this change: https://github.com/elysiajs/elysia/commit/61096846ecf37b481231de3c5ff754355313c903#diff-301d6ef66e0eb85d411eb60d0fc8122c687bdf6defeda688c48cf076084ec128R136

**Comment by ViliamKopecky** [comment]
Same issue

**Comment by fahreziadh** [comment]
Same issue here, i have double cookie when try to read via jwt plugin. the second plugin is being set with Session Expires

**Comment by l15c** [comment]
I'm having the same

**Comment by david-plugge** [comment]
Any chance this gets fixed soon? This issue does make elysia unusable for us.

**Comment by david-plugge** [comment]
Resolved in 1.1.22, you're goated
